source/slang/hlsl.meta.slang

```diff
@@ -2420,7 +2422,7 @@ extension __TextureImpl<T,Shape,isArray,0,sampleCount,0,isShadow,isCombined,form
                 {
                     if (isArray == 1)
                         // T read(uint2 coord, uint face, uint array, uint lod = 0) const
-                        __intrinsic_asm "$0.read(vec<uint,2>(($1).xy), uint(($1).z), uint(($1).w))";
+                        __intrinsic_asm "<invalid intrinsics>";
```

Comment:
There is now a static_cast intrinsic that we can use to report an error.

Reply:
Note about the solution: 
1. I have addressed the problem by adding `static_assert`.
2. I have kept `__intrinsic_asm` since I noticed else we get an stdlib compile error. The error seems to be due to the missing `__intrinsic_asm` (or an equivalent return statement).